config/consensus.go

```diff
@@ -422,6 +422,10 @@ var MaxEvalDeltaAccounts int
 // in a StateDelta, used for decoding purposes.
 var MaxStateDeltaKeys int
 
+// MaxLogCalls is the largest number of log calls that may appear in
+// an eval delta, used for decoding purposes.
+var MaxLogCalls int
```

Comment:
it should be consensus param since it is checked in `opLog` and a blind change would introduce compatibility problem.

Reply:
I disagree - these are the constants that we use only to bound message sizes.  This constant should *never* be used anywhere else, as these constants are allowed to be conservative, that is, it should be allowed to make this 37 and nothing bad would happen. (It would become possible to send txn objects with 37 log entries, but they should be illegal based on other rules, elsewhere.)

If that's *not* the intent - if this constant is used elsewhere to control the size (for example in the teal opcode), then it should be defined there, and this constant should be set separate.  There could be an argument for making that a consensus parameter rather than defined in logic package.  I can see both sides. But we generally know that any change in the way an opcode works is a consensus upgrade, and we gate them all on a single logicversion parameter, rather than define a bunch of separate ones.

Reply:
I'm fine with having it in`logic` similarly to max cumulative log size. But not the way it is done now - set a as a param and used in opcode.

Reply:
So I guess have two constants, and this one should be commented like: "Never decrease this value. It should be the highest allowable number of log messages allowed in any version"

This would be really weird, but I almost want to make it 33 so nobody thinks it's the right constant to use for anything else.

Reply:
this needs to be const, no need to initialize it `checkSetAllocBounds`.
```suggestion
const MaxLogCalls = 32
```

`checkSetAllocBounds` might have a check that the value is not greater than 32 (or 33 as JJ suggested)